test/integration/builder/inserts.js

```diff
@@ -1465,6 +1465,130 @@ module.exports = function (knex) {
       expect(rows[0].name).to.equal('AFTER');
     });
 
+    it('conditionally updates rows when inserting a duplicate key to unique column and merge with where clause matching row(s) is specified', async function () {
+      if (/redshift|mysql/i.test(knex.client.driverName)) {
```

Comment:
since you are asserting the error now, probably no longer you need to skip it

Reply:
Sorry about that. Fixed: 80f366d67fd90d4792470a3418668849c54daa36